dali/base/dadfs.cpp

```diff
@@ -6336,6 +6392,78 @@ class CDistributedSuperFile: public CDistributedFileBase<IDistributedSuperFile>
         return true;
     }
 
+    virtual bool getNumReads(stat_type &numReads)  const override
+    {
+        const IPropertyTree *attrs = root->queryPropTree("Attr");
+        if (attrs && attrs->hasProp(getDFUQResultFieldName(DFUQRFnumDiskReads)))
+            numReads = attrs->getPropInt64(getDFUQResultFieldName(DFUQRFnumDiskReads));
+        else
+        {
+            numReads = 0;
+            ForEachItemIn(i,subfiles)
+            {
+                stat_type c;
+                if (subfiles.item(i).getNumReads(c))
+                    numReads += c;
+            }
+        }
+        return numReads > 0;
+    }
+
+    virtual bool getNumWrites(stat_type &numWrites) const override
+    {
+        const IPropertyTree *attrs = root->queryPropTree("Attr");
+        if (attrs && attrs->hasProp(getDFUQResultFieldName(DFUQRFnumDiskWrites)))
+            numWrites = attrs->getPropInt64(getDFUQResultFieldName(DFUQRFnumDiskWrites));
+        else
+        {
+            numWrites = 0;
+            ForEachItemIn(i,subfiles)
+            {
+                stat_type c;
+                if (subfiles.item(i).getNumWrites(c))
+                    numWrites += c;
+            }
+        }
+        return numWrites > 0;
+    }
+
+    virtual bool getReadCost(cost_type &cost, bool calculateIfMissing=false) const override
```

Comment:
I may be missing where, but I can't see where calculateIfMissing is passed as true?

Reply:
It isn't at the moment. However, I think it should be by updateFileAttrs.  I've updated updateFileAttrs with calculateIfMissing->true for both getReadCost and getWriteCost.